Modules/getpath.c

```diff
@@ -447,7 +447,10 @@ getpath_realpath(PyObject *Py_UNUSED(self) , PyObject *args)
             if (s) {
                 *s = L'\0';
             }
-            path2 = _Py_normpath(_Py_join_relfile(path, resolved), -1);
+            path2 = _Py_join_relfile(path, resolved);
+            if (path2) {
+                path2 = _Py_normpath(path2, -1);
+            }
```

Comment:
If path2 is NULL, Don't we have to handle it as a raising an exception or do something else? Rather than passing to the `_Py_normpath`?

Reply:
This is already done at the end of the function by throwing a memory error